.github/workflows/api-compatibility.yml

```
@@ -0,0 +1,393 @@
+name: API Compatibility Check
+
+on:
+  pull_request:
+    branches: [ feature/v1-refactoring ]
+    paths:
+      - 'index.mjs'
+      - 'src/**/*.ts'
+      - 'src/**/*.js'
+      - 'package.json'
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  api-compatibility:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # Full history for comparison
+
+    - name: Use Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: '20.x'
+        cache: 'npm'
+
+    - name: Install dependencies
+      run: |
+        npm ci
+        # Install API analysis tools
+        npm list jsdoc || npm install --save-dev jsdoc@latest
+        npm list @apidevtools/json-schema-ref-parser || npm install --save-dev @apidevtools/json-schema-ref-parser@latest
+
+    - name: Extract Current API Surface
+      run: |
+        echo "🔍 Extracting current API surface from codebase..."
+        
+        # Create API extraction script
+        cat > extract-api.js << 'EOF'
+        const fs = require('fs');
+        const path = require('path');
+        
+        // Extract exports from index.mjs
+        function extractExports(filePath) {
+          if (!fs.existsSync(filePath)) return [];
+          
+          const content = fs.readFileSync(filePath, 'utf8');
+          const exports = [];
+          
+          // Find class definitions
+          const classMatches = content.match(/class\s+(\w+)/g);
+          if (classMatches) {
+            classMatches.forEach(match => {
+              const className = match.replace('class ', '');
+              exports.push({ type: 'class', name: className });
+            });
+          }
+          
+          // Find function definitions
+          const functionMatches = content.match(/(?:function\s+|const\s+\w+\s*=\s*(?:async\s+)?function|(?:async\s+)?function\s+)(\w+)/g);
+          if (functionMatches) {
+            functionMatches.forEach(match => {
+              const funcName = match.replace(/(?:function\s+|const\s+|=\s*(?:async\s+)?function|(?:async\s+)?function\s+)/g, '');
+              if (funcName && !funcName.includes('(')) {
+                exports.push({ type: 'function', name: funcName });
+              }
+            });
+          }
+          
+          // Find method definitions (simplified)
+          const methodMatches = content.match(/(\w+)\s*\([^)]*\)\s*{/g);
+          if (methodMatches) {
+            methodMatches.forEach(match => {
+              const methodName = match.replace(/\s*\([^)]*\)\s*{/, '');
+              if (methodName && !methodName.includes(' ') && methodName !== 'constructor') {
+                exports.push({ type: 'method', name: methodName });
+              }
+            });
+          }
```

The method extraction regex is too broad and will match function calls, not just method definitions. It needs to be scoped to class contexts:

```suggestion
          // Find method definitions within classes
          const classBlocks = content.match(/class\s+\w+\s*(?:extends\s+\w+\s*)?\{[^}]+\}/gs);
          if (classBlocks) {
            classBlocks.forEach(classBlock => {
              const methodMatches = classBlock.match(/(?:async\s+)?(\w+)\s*\([^)]*\)\s*\{/g);
              if (methodMatches) {
                methodMatches.forEach(match => {
                  const methodName = match.match(/(?:async\s+)?(\w+)\s*\(/)?.[1];
                  if (methodName && methodName !== 'constructor') {
                    exports.push({ type: 'method', name: methodName });
                  }
                });
              }
            });
          }
```